deps/ncrypto/ncrypto.cc

```diff
@@ -212,10 +212,25 @@ Buffer<void> DataPointer::release() {
 DataPointer DataPointer::resize(size_t len) {
   size_t actual_len = std::min(len_, len);
   auto buf = release();
-  if (actual_len == len_) return DataPointer(buf.data, actual_len);
-  buf.data = OPENSSL_realloc(buf.data, actual_len);
```

Comment:
This reallocation cannot realistically fail because of line 213: actual_len is min(len, len_), i.e., it always shrinks, never grows the buffer. 

I'd have suggested adding a `CHECK_NOT_NULL(buf.data)` sanity check but because this is code that's been moved from src/ to deps/ncrypto, you can't use that macro :facepalm: